hazelcast/src/main/java/com/hazelcast/client/impl/connection/tcp/TcpClientConnection.java

```diff
@@ -70,32 +69,35 @@ public class TcpClientConnection implements ClientConnection {
     private final long startTime = System.currentTimeMillis();
     private final Consumer<ClientMessage> responseHandler;
     private final ConcurrentMap attributeMap;
+    private final ChannelInitializer channelInitializer;
```

Comment:
It was volatile because we were setting it after constituting the connection object. I made it final, and set it while constructing the connection object